.github/workflows/nightly-ci-multiprocess-gpu.yml

```diff
@@ -1,77 +1,99 @@
-          source $JOBSCRIPTSDIR/slurm_utils_common.sh
-          sbatch -N 2 $JOBSCRIPTSDIR/multinode_pytest_jaxlib_nightly.sub | tee output.log
-          sleep 2m
-          export SLURM_JOBID=$(grep 'Submitted batch job' "output.log" | awk '{ print $4 }')
-          export SLURM_OUTPUT=$(scontrol show job "${SLURM_JOBID}" | grep 'StdOut' | awk -F '=' '{ print $2 }')
-          job_wait "${SLURM_JOBID}" & PID=$!
-          touch "${SLURM_OUTPUT}"
-          echo -e " ---------------------------------------------------\n" \
-               "----------WAITING FOR SLURM JOB TO BEGIN-----------\n" \
-               "---------------------------------------------------\n"
-          tail --pid="${PID}" -f "${SLURM_OUTPUT}"
-          export SLURM_STATE=$(job_state "${SLURM_JOBID}"); echo "SLURM_JOBID=${SLURM_JOBID} SLURM_STATE='${SLURM_STATE}'"
-          export SLURM_WALLTIME=$(job_time "${SLURM_JOBID}"); echo "SLURM_WALLTIME=${SLURM_WALLTIME} secs"
-          export SLURM_EXITCODE=$(job_exit_code "${SLURM_JOBID}" || echo $?); echo "SLURM_EXITCODE='${SLURM_EXITCODE}'"
-          if [ "${SLURM_EXITCODE}" != "0" ];      then exit ${SLURM_EXITCODE:-999}; fi
-          if [ "${SLURM_STATE}" != "COMPLETED" ]; then exit 1; fi
-      - name: Publish Test Results
-        uses: EnricoMi/publish-unit-test-result-action@46ab8d49369d898e381a607119161771bc65c2a6 # ratchet:EnricoMi/publish-unit-test-result-action@v2
+
+          # Create SSH keys
+          mkdir -p $GITHUB_WORKSPACE/.ssh && chmod 700 $GITHUB_WORKSPACE/.ssh
+          ssh-keygen -N '' -f $GITHUB_WORKSPACE/.ssh/id_rsa
+          PUBKEY=$(cat $GITHUB_WORKSPACE/.ssh/id_rsa.pub)
+
+          # Setup cluster, get IP address of head node, create SSH config
+          pip install -r "${GITHUB_WORKSPACE}/.github/workflows/slurm_job_scripts/requirements.txt"
+          python3 ${GITHUB_WORKSPACE}/.github/workflows/slurm_job_scripts/oci_cluster_manager.py create_cluster --pubkey "${PUBKEY}" &> ${GITHUB_WORKSPACE}/oci_automation_create.log
+          USER=$(tail -n 2 ${GITHUB_WORKSPACE}/oci_automation_create.log | head -n 1)
+          IP=$(tail -n 1 ${GITHUB_WORKSPACE}/oci_automation_create.log)
+          echo "::add-mask::${IP}"
+
+          grep "^${IP} " ${GITHUB_WORKSPACE}/oci_automation_create.log >> ${GITHUB_WORKSPACE}/.ssh/known_hosts
+          echo "Host headnode
+            User ${USER}
+            HostName ${IP}
+            IdentityFile ${GITHUB_WORKSPACE}/.ssh/id_rsa" > ${GITHUB_WORKSPACE}/.ssh/config
+
+          # SSH into the cluser & check SLURM
+          ssh ${CONFIG} headnode sinfo
+
+          # Run dummy job
+          ssh ${CONFIG} headnode 'srun --container-name=nvidia --container-image=docker://nvcr.io#nvidia/tensorflow:22.09-tf2-py3 -N 2 -t 15:00 --gpus-per-node=8 --cpus-per-task=8 --ntasks-per-node=8 timeout -v -k 0.5m 13m bash -c "hostname && nvidia-smi --query-gpu=gpu_name,driver_version --format=csv"'
```

Comment:
Can you break this into multiple lines? Quotes or here doc should work I think: https://stackoverflow.com/questions/32082646/write-ssh-command-over-multiple-lines

Reply:
I broke it down into a few pieces, should be better now.